### PR TITLE
Feature/2121 - Activity - Comment - Media/document/video Enhancement 

### LINKED ISSUE #2121: Activity - Comment - Media/document Enhancement
**Is your feature request related to a problem? Please describe.**
Currently whenever you upload a any Media/Document into the Activity comment it appearing into the user & groups Photos/Documents (Users/Groups )  page and also in main Photos & Documents directory page which is incorrect.

**Describe the solution you'd like**
Media/Documents uploaded into the Activity comment should not appear into the Photos/Documents page (Users/Groups ) and also in main Photos & Documents directory page.

**Describe alternatives you've considered**

- [ ] Remove the logic to create a seprate activity whenever you uploading a Media/Documents into the Activity comment.
- [ ] Creatre a Migration script which will run on update of platform and follow the below process on migration script.
  - [ ] 1. Query the Activity table to fetch all the activity  which have a **type**  `activity_comment`
  - [ ] 1.1. Loop the query result and find the meta `bp_document_ids` OR `bp_media_ids` respectively.
  - [ ] 1.1.1 Now loop the `bp_document_ids` OR `bp_media_ids`
  - [ ] 1.1.2 Now make the query to `media` OR `document` table respectively and delete the activity from those Media/Document and set the `activity_id` to `0` & `privacy` to `comment`
- [ ] Now on Activity ( Main, User and group ) page on activity comment Media/Document popup we have to make some changes to show the popup same as `Message` & `Forum`  

**Support ticket links**
If applicable, add HelpScout link or ticket number where the issue was originally reported.

### PR DESCRIPTION
**Pull request by jitendrabanjara1991**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

For this I have created hierarchical array and based on that migrate comment.

Fixes #2121.

### How to test the changes in this Pull Request:

1. First you need to add multiple comments with multiple replies with media and documents with latest release branch
2. Add comment above added media and document, and also add multiple replies there ( Check media/document in Photos/Documents Dir ) - https://screencast-o-matic.com/watch/crjXXCVj3jZ
2. After that move this branch and go to the Tools - Repair Community -> Repair Media Comments on the Activity/News Feed -> checked it -> Click on Repair Items
3. Media/Doc comment should not be display in Photos/Documents Dir

### Proof Screenshots or Video

https://screencast-o-matic.com/watch/crjXXvVj31B

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

Dependent API
https://github.com/buddyboss/buddyboss-platform-api/pull/314
Dependent App Native
https://github.com/buddyboss/buddyboss-app-native/issues/1565

### CODE REVIEW

**Comment by jitendrabanjara1991** [comment]
Hello @chetansatasiya 
I have completed media and document migration task.
Here is the video - https://screencast-o-matic.com/watch/cre0qvVVbhG
Please check it and let me know if anything else.
Thanks.

File: src/bp-activity/bp-activity-filters.php [hunk 1]
```diff
@@ -2360,25 +2043,489 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
...
+	return $repair_list;
+}
+/**
+ * Activity media migration process.
+ */
+function bb_activity_media_document_migration_process() {
+	global $wpdb;
+	$bp                  = buddypress();
+	$activity_table_name = $bp->activity->table_name;
+	$get_offset          = filter_input( INPUT_POST, 'offset', FILTER_SANITIZE_NUMBER_INT );
```
Comment (mehulkaklotar) [inline src/bp-activity/bp-activity-filters.php:2088]:
```suggestion
	$offset          = filter_input( INPUT_POST, 'offset', FILTER_VALIDATE_INT );
```

File: src/bp-activity/bp-activity-filters.php [hunk 2]
```diff
@@ -2360,25 +2043,489 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
...
+	$offset              = isset( $get_offset ) ? (int) ( $get_offset ) : 0;
```
Comment (mehulkaklotar) [inline src/bp-activity/bp-activity-filters.php:2089]:
```suggestion
```

File: src/bp-activity/bp-activity-filters.php [hunk 3]
```diff
@@ -2360,25 +2043,489 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
...
+	if ( 1 === $offset ) {
+		$offset = 0;
+	} else {
+		$offset = $offset;
+	}
+	$get_active_post_types = get_post_types( array( 'public' => true ) );
```
Comment (mehulkaklotar) [inline src/bp-activity/bp-activity-filters.php:2095]: can we change the variable name `$get_active_post_types` to `$active_post_types` ? 

Whenever we get data from function which name suggests it is getting data, we just need that data's name not get,fetch,update etc in the variable name.

File: src/bp-activity/bp-activity-filters.php [hunk 4]
```diff
@@ -2360,25 +2043,489 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
...
+			);
+		}
+	} else {
+		return array(
+			'status'  => 1,
+			'message' => __( 'media comment migration complete!', 'buddyboss' ),
+		);
+	}
+}
+/**
```
Comment (mehulkaklotar) [inline src/bp-activity/bp-activity-filters.php:2200]: add empty line between two functions and please add proper commenting on each important line of codes what it does and describe it.

File: src/bp-activity/bp-activity-filters.php [hunk 5]
```diff
@@ -2360,25 +2043,489 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
...
+			return;
+		}
+		$activity_id_store[] = $children_data->id;
+		$args                = array(
+			'display_comments' => true,
+			'show_hidden'      => true,
+			'sort'             => 'ASC',
+			'activity_ids'     => $children_data->id,
+		);
+		$get_activity_data   = bp_activity_get_specific( $args );
```
Comment (mehulkaklotar) [inline src/bp-activity/bp-activity-filters.php:2232]: here too, we don't need `get` in variable name, eg. `$get_activity_data` -> `$activity_data`

File: src/bp-activity/bp-activity-filters.php [hunk 6]
```diff
@@ -2360,25 +2043,489 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
...
+ * @param int    $secondary_item_id Secondary item id of the comment.
+ *
+ * @param string $comment_type      Type of the comment.
+ *
+ * @param string $comment_privacy   Privacy of the comment.
+ *
+ * @param int    $main_root_id      Main root comment id.
+ *
+ * @uses bb_activity_update_data() Use this function for the update data based data.
+ */
```
Comment (mehulkaklotar) [inline src/bp-activity/bp-activity-filters.php:2414]: why do we need empty lines in between param docs?

**Review by mehulkaklotar — CHANGES_REQUESTED** [review]
@jitendrabanjara1991 please fix all the suggestions and again put it in review. I will review again once you finish your changes.

**Comment by chetansatasiya** [comment]
@jitendrabanjara1991 This is not working as expected as we test on a call with the group activity. So need to make sure it will work with every case.

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Migration has a issue so it's not migrating with group activity

**Comment by jitendrabanjara1991** [comment]
@chetansatasiya 
Fixed - group migration issue. - https://screencast-o-matic.com/watch/crjY29VjJvC
Right now I have added migration code in repair community section. So let me know it need to move with background process.
Thanks.

**Comment by shakilAshraful** [comment]
@jitendrabanjara1991 My initial impression on this is that this issue **doesn't need any App Native change** as long as the fetch APIs (for medias, documents, videos) don't include medias/documents/videos posted with activity comments. This will most probably need API changes, but shouldn't require any App Native change.

cc. @KartikSuthar @chetansatasiya @sourabhmatolia

**Comment by jitendrabanjara1991** [comment]
We found another solution for this issue. So, we have created new PR for the same - https://github.com/buddyboss/buddyboss-platform/pull/2844. 
**What I have changed code in this PR?** - User will able to see comment for media/documents. Not able to do like/comment for the media. If user will add new media in child comment then user can not able to do like/comment.
So I am closing this PR now and will update on new PR.
cc - @firozwp @sourabhmatolia @KartikSuthar @chetansatasiya 
Thanks.

**Pull request closed without merging**